perllib/FixMyStreet/App/Form/Waste/Bulky.pm

```diff
@@ -76,20 +110,126 @@ has_field resident => (
     ],
 );
 
-has_field chosen_date => (
-    type => 'Select',
-    widget => 'RadioGroup',
-    required => 1,
-    label => 'Available dates',
-    options => [
-        # XXX look these up dynamically
-        { value => '20220825', label => '2022-08-25' },
-        { value => '20220826', label => '2022-08-26' },
-        { value => '20220827', label => '2022-08-27' },
-        { value => '20220828', label => '2022-08-28' },
-    ],
+has_field chosen_date_earlier => (
+    _chosen_date_settings(),
+    options_method => sub {
+        my $self = shift;
+        my $form = $self->form;
+        my $c    = $form->c;
+
+        my @dates = _get_dates($c);
+
```

Comment:
This blocks the page moving forwards while it loads - so the loading happens on the about_you page. It may be a minor thing, but I think it would make sense to move forward to the choose_date_earlier and for loading to take place there. How? Not entirely sure...

Reply:
We could potentially do an asynchronous fetch of data via JS, but that may be outside the scope of the (more-or-less) minimum viable product we are aiming for (it also doesn't really solve the problem of the API calls being verrrry sloooow - I may need to look at the parallelisation work that Dave did for other calls).